### PR TITLE
Refactor create election command

### PR DESCRIPTION
**Pull request by leio10**
This PR prepares the bulletin to allow to parse JWT tokens including validations of the sender signature. To do that, we need to have the public keys of the authority, the BB and the trustees in the `create_election` message.

This first refactor the gem settings, to simplify them, as the quorum will be a core settings of the system, and not a voting scheme parameter.

Then, it also refactors the client to remove all global instances to improve the maintainability and ease the testing process.

Finally, it moves the message building for the `create_election` command from Decidim into the gem, requiring to pass a very simple hash to the command, with only seven root keys: `trustees`, `default_locale`, `title`, `start_date`, `end_date`, `questions` and `answers`.

### CODE REVIEW

**Review by agustibr — APPROVED** [review]
Great! Keeps improving 🌟

File: decidim-bulletin_board-ruby/lib/decidim/bulletin_board/client.rb [hunk 1]
```diff
@@ -1,109 +1,102 @@
...
-      def initialize
-        @server = BulletinBoard.server.presence
-        @api_key = BulletinBoard.api_key.presence
-        @scheme = BulletinBoard.scheme.presence
-        @authority_name = BulletinBoard.authority_name.presence
-        @number_of_trustees = BulletinBoard.number_of_trustees.presence
-        @server_public_key = BulletinBoard.server_public_key.presence
-        @identification_private_key = BulletinBoard.identification_private_key.presence
-        @private_key = identification_private_key_content if identification_private_key
+      def initialize(config = Decidim::BulletinBoard)
```
Comment (slickepinne) [inline decidim-bulletin_board-ruby/lib/decidim/bulletin_board/client.rb:22]: Nice 😊

File: decidim-bulletin_board-ruby/CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Changed
 
 - The `get_election_log_entries_by_type` command got renamed to `get_election_results`.
+- The `create_election` command now receives all the information needed in a simple hash and builds the message to add to the election log. It only needs seven main keys: `trustees`, `default_locale`, `title`, `start_date`, `end_date`, `questions` and `answers`.
+- The `scheme` setting was replaced by the `scheme_name` and `quorum` settings.
```
Comment (slickepinne) [inline decidim-bulletin_board-ruby/CHANGELOG.md:19]: Not sure if we should mention the renaming of `client` into `graphql`?

**Review by slickepinne — APPROVED** [review]
Great! 😊

File: decidim-bulletin_board-ruby/CHANGELOG.md (hunk 2 above)
Comment (leio10) [reply decidim-bulletin_board-ruby/CHANGELOG.md:19]: I think that's an internal thing, I understand that the changelog is to document the changes done in the public interface.

**Review by beagleknight — APPROVED** [review]
Oh wow, great refactor ✨

**Pull request merged**